### PR TITLE
JS POST OOP

### PR DESCRIPTION
**Pull request by tavor118sn**
### JS POST OOP

[Demo](https://tavor118sn.github.io/a-tiny-JS-world/)
[Code base](https://github.com/tavor118sn/a-tiny-JS-world/tree/master)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/tavor118sn/task_js_post_oop/index.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    this.saying = say;
+    this._friends = friends
+  }
+
+  get friends() {
+    let friendsRepresentation = '';
+    if (this._friends && this._friends.length > 0) {
+      friendsRepresentation = this._friends.join(', ');
+    }
+    return friendsRepresentation;
```
Comment (babayK0) [inline submissions/tavor118sn/task_js_post_oop/index.js:28]: Why not just return with ternary operator ? 

```suggestion
    return condition ? this._friends.join(', ') : '';
```

File: submissions/tavor118sn/task_js_post_oop/index.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    this.age = age;
+    this.gender = gender;
+    this.legs = legs;
+    this.saying = say;
+    this._friends = friends
+  }
+
+  get friends() {
+    let friendsRepresentation = '';
+    if (this._friends && this._friends.length > 0) {
```
Comment (babayK0) [inline submissions/tavor118sn/task_js_post_oop/index.js:25]:
```suggestion
    if (this._friends && this._friends.length) {
```
You can do check like this

File: submissions/tavor118sn/task_js_post_oop/index.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    ]
+  }
+
+  toString() {
+    return this.getAllCharacteristics().join('; ')
+  }
+}
+
+class Human extends Creature {
+  constructor(name, age, gender, saying, friends) {
```
Comment (babayK0) [inline submissions/tavor118sn/task_js_post_oop/index.js:49]: Why i can't pass hands and legs while initializing ?

File: submissions/tavor118sn/task_js_post_oop/index.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    let species = 'human';
+    let legs = 2;
```
Comment (babayK0) [inline submissions/tavor118sn/task_js_post_oop/index.js:51]: Try to implement with [default parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters)

File: submissions/tavor118sn/task_js_post_oop/index.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+class Woman extends Human {
+  constructor(name, age, saying, friends) {
+    let gender = 'female';
+    super(name, age, gender, saying, friends);
+  }
+}
+
+class Animal extends Creature {
+  constructor(species, name, age, gender, say, friends) {
+    let legs = 4;
```
Comment (babayK0) [inline submissions/tavor118sn/task_js_post_oop/index.js:79]: Not all animals have legs

**Review by babayK0 — CHANGES_REQUESTED** [review]
Good job!
Nice implementatiion of `toString` =)
I left some comments below)

**Comment by tavor118sn** [comment]
Thanks for review! Hope I understood correctly.

File: submissions/tavor118sn/task_js_post_oop/index.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+  getAllCharacteristics() {
+    return [
+      `${this.species}`,
+      `<strong>${this.name}</strong>`,
+      `${this.age}`,
+      `${this.gender}`,
+      `${this.legs}`,
+      `${this.saying}`,
+      `${this.friends}`,
```
Comment (OleksiyRudenko) [inline submissions/tavor118sn/task_js_post_oop/index.js:35]: Just wondering where this type of reference to variables (like `` `${varName}` ``) comes from? Please explain.
(`` `<strong>${this.name}</strong>` `` is out of the question)

Comment (tavor118sn) [reply submissions/tavor118sn/task_js_post_oop/index.js:35]: Sorry! Copy-pasted from my previous task solution. Will be fixed.

Comment (OleksiyRudenko) [reply submissions/tavor118sn/task_js_post_oop/index.js:35]: @tavor118sn doesn't really answer the question.
The question was not "where did you take this fragment of code from" but "where this **type of referencing** comes from".

Comment (OleksiyRudenko) [reply submissions/tavor118sn/task_js_post_oop/index.js:35]: I see. We have already discussed this topic in one of your past tasks.
See, how anti-patterns draw attention.

**Review by babayK0 — APPROVED** [review]
LGTM!

@OleksiyRudenko i am needed your approval

**Review by OleksiyRudenko — APPROVED** [review]
A bit discriminating hierarchy where humans occur to be worth a dedicated class and other animals do not, but technically it works. ;)
@tavor118sn well done!

**Pull request merged**